File: gpt-3.5/one-shot/acc_attach.c
#ifndef T1

int test1(){
    int err = 0;

    #pragma acc enter data create(a[0:10])
    #pragma acc attach(a[0:2])

    for (int i = 0; i < 2; i++)
        a[i] = i;

    #pragma acc exit data delete(a[0:2])
    #pragma acc exit data copyout(a[0:10])

    for (int i = 0; i < 10; i++){
        if (a[i] != i){
            err++;
            break;
        }
    }

    return err;
}

#endif

int main(){
    int failcode = 0;
    int failed;
#ifndef T1
    failed = 0;
    for (int x = 0; x < NUM_TEST_CALLS; ++x){
        failed = failed + test1();
    }
    if (failed != 0){
        failcode = failcode + (1 << 0);
    }
#endif
    return failcode;
}

File: bad-tests-v2/acc_attach.f90
#ifndef T1
!T1:acc attach,V:2.7-2.3
      LOGICAL FUNCTION test1()
        USE OPENACC
        IMPLICIT NONE
        INCLUDE "acc_testsuite.Fh"
        
        INTEGER :: errors = 0
        INTEGER, DIMENSION(10) :: host_array, device_array
        TYPE(C_PTR) :: host_ptr, device_ptr

        host_array = 1
        device_array = 0

        host_ptr = C_LOC(host_array)
        device_ptr = C_LOC(device_array)

        ! Attach device_array to host_array
        CALL acc_attach_ptr(host_ptr, device_ptr, 10 * 4)

        ! Check if device_array is updated
        IF (ANY(device_array .NE. host_array)) THEN
          errors = errors + 1
        END IF

        ! Detach device_array from host_array
        CALL acc_detach_ptr(host_ptr)

        IF (errors .eq. 0) THEN
          test1 = .FALSE.
        ELSE
          test1 = .TRUE.
        END IF
      END FUNCTION test1
#endif

      PROGRAM main
        IMPLICIT NONE
        INTEGER :: failcode, testrun
        LOGICAL :: failed
        INCLUDE "acc_testsuite.Fh"
#ifndef T1
        LOGICAL :: test1
#endif
        failed = .FALSE.
        failcode = 0
#ifndef T1
        DO testrun = 1, NUM_TEST_CALLS
          failed = failed .or. test1()
        END DO
        IF (failed) THEN
          failcode = failcode + 2 ** 0
          failed = .FALSE.
        END IF
#endif
        CALL EXIT (failcode)
      END PROGRAM main